### DISCUSSION TITLE
sp-sidenav-item

### DISCUSSION THREAD

**Discussion by rinaok** [adobe/spectrum-web-components#2398] [Q&A] [answered]
Hello,

My question is about the implementation of the sp-sidenav-item component:

The sp-sidenav-item acts both as a child and as a parent. 
An **anchor** element has a href attribute, which is fine for the child element, but a bit problematic for the parent element. 
Since the parent's role is to toggle its children, it should be a **Button** rather than an **Anchor**.

Changing the parent component to be the [**Details disclosure element**](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) and the child component to be the [**Disclosure Summary element**](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary) would make sense to me.

I would appreciate your thoughts on this.

Thank you!

**Comment by Westbrook** [comment, marked as answer]
While it might not be common for a parent to have an `href` it can. This is particularly useful when you have a landing page for a section. When this is the case, the benefit of having the current structure is much more clear.

While the `details/summary` relationship is quite interesting, it's hard to say that it would be discreetly better for this use case. Particularly, the `<summary>` element is not to kind of `<a>` elements being included within it, and in the case that a user had a "collection" page to surface there would be some difficulties created by adopting this structure.

**Reply by rinaok** [reply]
why would a disclosing button contain `<a>` to begin with?

**Reply by Westbrook** [reply]
Because it isn’t _only_ a disclosure button. While it can be just a disclosure button, the currently delivered functionality allows for the presence of a link there as well. In this case the children would open and new content would be come available at the same time.

We don’t currently exhibit this behavior in our site or our demos, but we should find a useful way to add that. However, not wanting to remove that capability leads to the benefits of persisting the current structure.

**Reply by rinaok** [reply]
I understand the intent, and now I realize the above question is confusing. Just curious if an anchor should play a role of a collapse/expand button.

**Reply by Westbrook** [reply]
I don’t personally see why not. It might not be the best choice in an MPA, but definitely in an SPA it makes a lot of sense. I’d accept arguments against if you’ve got good ones.
